presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -286,13 +286,6 @@ void PrestoServer::run() {
   facebook::velox::exec::ExchangeSource::registerFactory(
       operators::UnsafeRowExchangeSource::createExchangeSource);
 
```

Comment:
@majetideepak : Unrelated to this PR. Java dissociates connectors and readers. This might come up during ConnectorSPI discussion, but that might be a route to consider.

Reply:
@aditi-pandit The connectors and readers are still disassociated in Prestissimo/Velox. The recent change https://github.com/facebookincubator/velox/pull/5469 in Velox simplifies the registration of existing Velox native readers, writers, file-system adapters by registering them along with the HiveConnector instance instead of at the client like Prestissimo.
This avoids code duplication at all such clients as we can see in this PR and on the Velox side.
In a hypothetical scenario where someone wants to use a custom ParquetReader, they can unregister the native ParquetReader and register the custom one.

Reply:
@majetideepak : It might be more clear to add a registerDefaultReadersAndWriters/registerDefaultFileSystems function in Velox and invoke that in all the engines, so that the registration step is more explicit I feel.  The engine's can then have their own lists if they prefer another set.

Registering all and then unregistering some readers/writers makes it more error prone.

Reply:
In practice, nobody is going to come up with their own Parquet readers or S3FileSystem.
It is more likely that if someone is using the HiveConnector in their application, they would want all its default components Parquet, DWRF, S3, etc. to be installed already.
I will add a comment above the Hive Connector registration that these default components will be registered. That should clarify this further.

Reply:
Just to clarify, we do not register the velox::HiveConnector in Prestissimo. It is registered in Velox itself
https://github.com/facebookincubator/velox/blob/main/velox/connectors/hive/HiveConnector.cpp#L160
Again, from a user standpoint, I don't see why we should explicitly register them in Prestissimo.

Reply:
My initial comment came from the thought that the java and C++ Connector SPI and artifacts should be consistent. We can delve into this at a later time. For this PR the current philosophy is okay.

But from the point of view of this change then, its a bit odd that Velox HiveConnector registers S3Filesystem by default (guarded by the flags offcourse) but HDFS is optional.
Also, maybe Dwrf reader/writer registration should also be flag guarded.